From f958fdd4b1dfa4ffe829b4c0a2773be5ee2f94bf Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 20 Aug 2017 18:26:20 +0200
Subject: [PATCH 1/1] efi_loader: non-static efi_open_protocol,
 efi_close_protocol

We need efi_open_protocol and efi_close_protocol for
implementing other functions. So they shouldn't be static.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi_loader.h          | 9 +++++++++
 lib/efi_loader/efi_boottime.c | 9 ++++-----
 2 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/include/efi_loader.h b/include/efi_loader.h
index c4460c8986..51e7136ae0 100644
--- a/include/efi_loader.h
+++ b/include/efi_loader.h
@@ -199,6 +199,15 @@ efi_status_t efi_set_timer(struct efi_event *event, enum efi_timer_delay type,
 			   uint64_t trigger_time);
 /* Call this to signal an event */
 void efi_signal_event(struct efi_event *event);
+/* Call this with EFI_CALL to close a protocol */
+efi_status_t EFIAPI efi_close_protocol(void *handle, const efi_guid_t *protocol,
+				       void *agent_handle,
+				       void *controller_handle);
+/* Call this with EFI_CALL to open a protocol */
+efi_status_t EFIAPI efi_open_protocol(
+			void *handle, const efi_guid_t *protocol,
+			void **protocol_interface, void *agent_handle,
+			void *controller_handle, uint32_t attributes);
 
 /* open file system: */
 struct efi_simple_file_system_protocol *efi_simple_file_system(
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index 204aacfac1..4806e8f28c 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -1551,10 +1551,9 @@ static efi_status_t EFIAPI efi_disconnect_controller(void *controller_handle,
  * @controller_handle	handle of the controller
  * @return		status code
  */
-static efi_status_t EFIAPI efi_close_protocol(void *handle,
-					      const efi_guid_t *protocol,
-					      void *agent_handle,
-					      void *controller_handle)
+efi_status_t EFIAPI efi_close_protocol(void *handle, const efi_guid_t *protocol,
+				       void *agent_handle,
+				       void *controller_handle)
 {
 	struct efi_handler *handler;
 	size_t i;
@@ -1959,7 +1958,7 @@ efi_status_t EFIAPI efi_open_protocol(
  * @attributes		attributes indicating how to open the protocol
  * @return		status code
  */
-static efi_status_t EFIAPI efi_open_protocol(
+efi_status_t EFIAPI efi_open_protocol(
 			void *handle, const efi_guid_t *protocol,
 			void **protocol_interface, void *agent_handle,
 			void *controller_handle, uint32_t attributes)
-- 
2.14.1

